### PR TITLE
Add gRPC Authority connection option

### PR DESCRIPTION
**Pull request by alexshtin**
## What was changed:

Add gRPC Authority connection option.

## Why?

gRPC Authority is used for routing in certain scenarios. This is a safe change as gRPC enforces that authority can only be overridden in insecure mode.

### How was this tested
Integration and unit tests.

### CODE REVIEW

**Review by wxing1292 — APPROVED** [review]

File: internal/client.go
Comment (flossypurse) [inline internal/client.go:414]: "Only works" might imply an error might happen?

Is it "only used when TLS is nil" ?

File: internal/client.go
Comment (flossypurse) [inline internal/client.go:417]:
```suggestion
		// By default, after gRPC connection to  the Server is created, client will make a request to
```

File: internal/client.go
Comment (flossypurse) [inline internal/client.go:418]:
```suggestion
		// health check endpoint to make sure the Server is accessible.
```

File: internal/client.go
Comment (flossypurse) [inline internal/client.go:422]: Shouldn't the comment start with the Field name?

**Review by vitarb — APPROVED** [review]
Approved assuming Cully's concerns are addressed before landing.

File: internal/client.go
Comment (alexshtin) [reply internal/client.go:422]: It doesn't have too. But here it is easy to fix.

**Pull request merged**
